frontend/src/console/components/RedeploymentToolbar/RedeploymentToolbar.tsx

```diff
@@ -70,73 +70,78 @@ export const RedeploymentToolbar: React.FC<Props> = ({
+            redeploymentStatus={deploymentStatus}
+            cancelRedeployment={cancelDeploymentHandler}
           />
-        </Button>
-      </div>
 
-      <Modal
-        show={show}
-        onHide={handleCloseModal}
-        className="cancel-redeployment-modal"
-        centered
-      >
-        <Modal.Header>
-          <Modal.Title>
-            <p>
-              <WrappedMessage
-                id="cancelDeploymentConfirm"
-                messages={messages}
-              />
-            </p>
-          </Modal.Title>
-        </Modal.Header>
-        <Modal.Body>
-          <WrappedMessage
-            id="cancelDeploymentConfirmText"
-            messages={messages}
-          />
-        </Modal.Body>
-        <Modal.Footer>
-          <Button
-            variant="outline-primary"
-            size="lg"
-            onClick={handleCloseModal}
-          >
-            <WrappedMessage id="closeModalButton" messages={messages} />
-          </Button>
           <Button
+            className="float-right loading"
             variant="primary"
             size="lg"
             onClick={() => {
-              cancelRedeployment();
-              handleCloseModal();
+              performDeployment();
             }}
+            disabled={deploymentDisabled}
           >
-            <WrappedMessage id="confirmCancelDeployment" messages={messages} />
+            <WrappedMessage
+              id="deploy"
+              messages={messages}
+              values={{ undeployedChanges }}
+            />
           </Button>
-        </Modal.Footer>
-      </Modal>
+        </div>
+
+        <Modal
+          show={show}
+          onHide={handleCloseModal}
+          className="cancel-redeployment-modal"
+          centered
+        >
+          <Modal.Header>
+            <Modal.Title>
+              <p>
+                <WrappedMessage
+                  id="cancelDeploymentConfirm"
+                  messages={messages}
+                />
+              </p>
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <WrappedMessage
+              id="cancelDeploymentConfirmText"
+              messages={messages}
+            />
+          </Modal.Body>
+          <Modal.Footer>
+            <Button
+              variant="outline-primary"
+              size="lg"
+              onClick={handleCloseModal}
+            >
+              <WrappedMessage id="closeModalButton" messages={messages} />
+            </Button>
+            <Button
+              variant="primary"
+              size="lg"
+              onClick={() => {
+                cancelRedeployment();
+                handleCloseModal();
+              }}
+            >
+              <WrappedMessage
+                id="confirmCancelDeployment"
+                messages={messages}
+              />
+            </Button>
+          </Modal.Footer>
+        </Modal>
+      </div>
     </div>
   );
 };
```

Comment:
If there are no longer any changes in this file, please remove it from the commit.